.github/workflows/build-pip-packages.yml

```diff
@@ -45,10 +45,10 @@ jobs:
         run: |
           python3 -m pip install --upgrade pip
           python3 -m pip install build
+          python3 -m pip install toml
 
       - name: Update PYPI Version
-        working-directory: python
-        run: sed -i 's/version = .*/version = "${{ inputs.ice_version }}"/' pyproject.toml
+        run: python3 packaging/pypi/update_version.py ${{ inputs.ice_version }} python/pyproject.toml
```

Comment:
Added a small script that uses toml to parse and update the project version in the project file. This works better here as it runs in both Windows/macOS runners.